core/templates/pages/admin-page/jobs-tab/admin-jobs-tab.component.spec.ts

```diff
@@ -0,0 +1,354 @@
+    expect(componentInstance.AUDIT_JOB_SPECS)
+      .toEqual(mockAdminPageData.auditJobStatusSummaries);
+    expect(componentInstance.RECENT_JOB_DATA)
+      .toEqual(mockAdminPageData.recentJobData);
+    expect(componentInstance.showingJobOutput).toBeFalse();
+  });
+
+  it('should work as expected when showJobOutput is called', () => {
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
+    let element: HTMLElement = document.createElement('div');
+    spyOn(document, 'querySelector').and.returnValue(element);
+    spyOn(element, 'scrollIntoView');
+
+    componentInstance.showJobOutput(mockJobId);
+    expect(componentInstance.showingJobOutput).toBeTrue();
+    expect(componentInstance.jobOutput).toEqual(['1', '2', '3']);
+    expect(element.scrollIntoView).toHaveBeenCalled();
+  });
+
+  it('should except error when showJobOuput is called', () => {
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
+    spyOn(componentInstance.setStatusMessage, 'emit');
+    componentInstance.showJobOutput('error');
+    expect(componentInstance.setStatusMessage.emit)
+      .toHaveBeenCalledWith('Server error: test error');
+  });
+
+  it('should work as expected when startNewJob is called', () =>{
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
+    spyOn(componentInstance.setStatusMessage, 'emit');
+    componentInstance.startNewJob('testJob');
+    expect(componentInstance.setStatusMessage.emit)
+      .toHaveBeenCalledWith('Job started successfully.');
+  });
+
+  it('should expect error when startNewJob is called', () => {
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
+    spyOn(componentInstance.setStatusMessage, 'emit');
+    componentInstance.startNewJob('error');
+    expect(componentInstance.setStatusMessage.emit)
+      .toHaveBeenCalledWith('Server error: test error');
+  });
+
+  it('should start computation', () => {
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
+    spyOn(componentInstance.setStatusMessage, 'emit');
+    componentInstance
+      .startComputation('testComputation');
+    expect(componentInstance.setStatusMessage.emit)
+      .toHaveBeenCalledWith('Computation started successfully.');
+  });
+
+  it('should expect error when startCompuation is called', () => {
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
+    spyOn(componentInstance.setStatusMessage, 'emit');
+    componentInstance
+      .startComputation('error');
+    expect(componentInstance.setStatusMessage.emit)
+      .toHaveBeenCalledWith('Server error: test error');
+  });
+
+  it('should stop computation', () => {
+    componentInstance =
+    (componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
```

Comment:
```suggestion
    componentInstance = (
      componentInstance as unknown) as jasmine.SpyObj<AdminJobsTabComponent>;
```
Break after the parenthesis, here and elsewhere!

Reply:
Done.